docs/Tutorials/Privacy/Private-Transaction-Performance-Best-Practice.md

```diff
@@ -0,0 +1,73 @@
+This limits the throughput to at most one private transaction per block when submitting from a single thread.
+To increase throughput, use web3js-quorum from multiple concurrent threads or processes.
+
+### Colocate Besu and Tessera
+
+Besu has to talk to its local Tessera node frequently while handling a block.
+While it is recommended to not run them on the same node, one should minimize the latency between Besu and Tessera to improve block processing times.
+Besu and Tessera should not be hosted in geographically distributed locations.
+
+### Optimize worst-case latency between Tessera nodes
+
+When distributing a new private transaction between Tessera nodes, the overall throughput is determined by the slowest Tessera nodes.
+Try to minimize network latency between Tessera nodes and do not mix different machine types when hosting Tessera.
+
+### Use stateful nonce management
+
+Management of public and private nonces in web3js-quorum is stateless: before a transaction is sent, web3js-quorum has to query for those nonces.
+This is increasing latency, the node's load, and is a source of fragility due to nonce collision when multiple senders try to use the same account concurrently.
+For performance and reliability it is advantageous to statefully manage nonces on the client side instead of querying them for every transaction.
+If custom code for this is not an option, Orchestrate can be used.
+
+### Use random PMT senders
+
+To avoid public nonce management, private marker transaction can be send using a [random account per transaction](https://besu.hyperledger.org/en/stable/Reference/CLI/CLI-Syntax/#privacy-marker-transaction-signing-key-file).
+This option is only available for zero gas networks.
+
+### Avoid queuing transactions in Tessera
+
+When Tessera is overloaded with transactions, the performance breaks down catastrophically due to unbounded growth of the request queue.
+Avoid sending more transactions to Tessera than it can handle.
+Sudden jumps in submission latency and submission failure rate should be answered with a load reduction on the client side, for example using a backoff scheme.
+
+Please note that this is not Tessera specific but a general issue in distributed systems.
+It just happens that if queueing discipline is not maintained, Tessera tends to be the first components to fail.
+
+### Limit the group size to reduce communication overhead
+
+Smaller groups need fewer communication for transaction propagation.
+If reducing the number of Tessera nodes involved in a transaction is an option, it will lead to slightly better tail latencies.
+Multi-tenancy Tessera can be used to have large groups with a small number of Tessera nodes–possibly only one.
+
+### Limit group membership changes and make them quick
+
+Groups are locked (that is barred from executing transactions) during membership changes.
```

Comment:
s/that is barred from executing transactions/prevented from executing transactions